Traceback (most recent call last):
  File "/Users/johannes/opt/anaconda3/envs/books/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/Users/johannes/opt/anaconda3/envs/books/lib/python3.8/site-packages/nbclient/client.py", line 1087, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/johannes/opt/anaconda3/envs/books/lib/python3.8/site-packages/nbclient/util.py", line 74, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/Users/johannes/opt/anaconda3/envs/books/lib/python3.8/site-packages/nbclient/util.py", line 53, in just_run
    return loop.run_until_complete(coro)
  File "/Users/johannes/opt/anaconda3/envs/books/lib/python3.8/asyncio/base_events.py", line 608, in run_until_complete
    return future.result()
  File "/Users/johannes/opt/anaconda3/envs/books/lib/python3.8/site-packages/nbclient/client.py", line 540, in async_execute
    await self.async_execute_cell(
  File "/Users/johannes/opt/anaconda3/envs/books/lib/python3.8/site-packages/nbclient/client.py", line 832, in async_execute_cell
    self._check_raise_for_error(cell, exec_reply)
  File "/Users/johannes/opt/anaconda3/envs/books/lib/python3.8/site-packages/nbclient/client.py", line 740, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply['content'])
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
train(30, 128)
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mAttributeError[0m                            Traceback (most recent call last)
[0;32m<ipython-input-17-57f9cef8857f>[0m in [0;36m<module>[0;34m[0m
[0;32m----> 1[0;31m [0mtrain[0m[0;34m([0m[0;36m30[0m[0;34m,[0m [0;36m128[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m
[0;32m<ipython-input-16-036338951789>[0m in [0;36mtrain[0;34m(epoch, batch_size)[0m
[1;32m     22[0m             [0;31m# train the discriminator[0m[0;34m[0m[0;34m[0m[0;34m[0m[0m
[1;32m     23[0m             [0mdiscriminator[0m[0;34m.[0m[0mtrainable[0m [0;34m=[0m [0;32mTrue[0m[0;34m[0m[0;34m[0m[0m
[0;32m---> 24[0;31m             [0mdiscriminator[0m[0;34m.[0m[0mtrain_on_batch[0m[0;34m([0m[0mX[0m[0;34m,[0m [0my_discriminator[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m     25[0m [0;34m[0m[0m
[1;32m     26[0m             [0;31m# train the generator[0m[0;34m[0m[0;34m[0m[0;34m[0m[0m

[0;32m~/opt/anaconda3/envs/books/lib/python3.8/site-packages/tensorflow/python/keras/engine/training.py[0m in [0;36mtrain_on_batch[0;34m(self, x, y, sample_weight, class_weight, reset_metrics, return_dict)[0m
[1;32m   1721[0m     [0;32mwith[0m [0mself[0m[0;34m.[0m[0mdistribute_strategy[0m[0;34m.[0m[0mscope[0m[0;34m([0m[0;34m)[0m[0;34m,[0m[0;31m [0m[0;31m\[0m[0;34m[0m[0;34m[0m[0m
[1;32m   1722[0m          [0mtraining_utils[0m[0;34m.[0m[0mRespectCompiledTrainableState[0m[0;34m([0m[0mself[0m[0;34m)[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[0;32m-> 1723[0;31m       iterator = data_adapter.single_batch_iterator(self.distribute_strategy, x,
[0m[1;32m   1724[0m                                                     [0my[0m[0;34m,[0m [0msample_weight[0m[0;34m,[0m[0;34m[0m[0;34m[0m[0m
[1;32m   1725[0m                                                     class_weight)

[0;32m~/opt/anaconda3/envs/books/lib/python3.8/site-packages/tensorflow/python/keras/engine/data_adapter.py[0m in [0;36msingle_batch_iterator[0;34m(strategy, x, y, sample_weight, class_weight)[0m
[1;32m   1511[0m     [0mdata[0m [0;34m=[0m [0;34m([0m[0mx[0m[0;34m,[0m [0my[0m[0;34m,[0m [0msample_weight[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[1;32m   1512[0m [0;34m[0m[0m
[0;32m-> 1513[0;31m   [0m_check_data_cardinality[0m[0;34m([0m[0mdata[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m   1514[0m   [0mdataset[0m [0;34m=[0m [0mdataset_ops[0m[0;34m.[0m[0mDatasetV2[0m[0;34m.[0m[0mfrom_tensors[0m[0;34m([0m[0mdata[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[1;32m   1515[0m   [0;32mif[0m [0mclass_weight[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m

[0;32m~/opt/anaconda3/envs/books/lib/python3.8/site-packages/tensorflow/python/keras/engine/data_adapter.py[0m in [0;36m_check_data_cardinality[0;34m(data)[0m
[1;32m   1520[0m [0;34m[0m[0m
[1;32m   1521[0m [0;32mdef[0m [0m_check_data_cardinality[0m[0;34m([0m[0mdata[0m[0;34m)[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[0;32m-> 1522[0;31m   [0mnum_samples[0m [0;34m=[0m [0mset[0m[0;34m([0m[0mint[0m[0;34m([0m[0mi[0m[0;34m.[0m[0mshape[0m[0;34m[[0m[0;36m0[0m[0;34m][0m[0;34m)[0m [0;32mfor[0m [0mi[0m [0;32min[0m [0mnest[0m[0;34m.[0m[0mflatten[0m[0;34m([0m[0mdata[0m[0;34m)[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m   1523[0m   [0;32mif[0m [0mlen[0m[0;34m([0m[0mnum_samples[0m[0;34m)[0m [0;34m>[0m [0;36m1[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[1;32m   1524[0m     [0mmsg[0m [0;34m=[0m [0;34m"Data cardinality is ambiguous:\n"[0m[0;34m[0m[0;34m[0m[0m

[0;32m~/opt/anaconda3/envs/books/lib/python3.8/site-packages/tensorflow/python/keras/engine/data_adapter.py[0m in [0;36m<genexpr>[0;34m(.0)[0m
[1;32m   1520[0m [0;34m[0m[0m
[1;32m   1521[0m [0;32mdef[0m [0m_check_data_cardinality[0m[0;34m([0m[0mdata[0m[0;34m)[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[0;32m-> 1522[0;31m   [0mnum_samples[0m [0;34m=[0m [0mset[0m[0;34m([0m[0mint[0m[0;34m([0m[0mi[0m[0;34m.[0m[0mshape[0m[0;34m[[0m[0;36m0[0m[0;34m][0m[0;34m)[0m [0;32mfor[0m [0mi[0m [0;32min[0m [0mnest[0m[0;34m.[0m[0mflatten[0m[0;34m([0m[0mdata[0m[0;34m)[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m   1523[0m   [0;32mif[0m [0mlen[0m[0;34m([0m[0mnum_samples[0m[0;34m)[0m [0;34m>[0m [0;36m1[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[1;32m   1524[0m     [0mmsg[0m [0;34m=[0m [0;34m"Data cardinality is ambiguous:\n"[0m[0;34m[0m[0;34m[0m[0m

[0;31mAttributeError[0m: 'int' object has no attribute 'shape'
AttributeError: 'int' object has no attribute 'shape'

